Standards/scs-0217-v1-cluster-hardening.md

```diff
@@ -0,0 +1,470 @@
+certificates, since etcd by default trusts all the certificates issued by the root CA [2][nsa-cisa].
+More information about authentication via TLS are provided in the chapter [ACL restrictions](#acl-restrictions).
+
+### Securing endpoints
+
+Kubernetes provides a well-defined set of ports in its default configuration. These ports are
+used for inter-component communication as well as external access. Due to the distribution of
+and information about Kubernetes clusters, it is easy for a bad actor to identify a clusters
+ports and try to attack them. In order to minimize the attack vector, internal ports (and therefore components)
+should not be accessible from external networks, except if there are requirements to enable this behavior.
+
+A good way to restrict access would be a combination of firewalls with port
+blocking possible network separation.
+How this is done is highly dependent on the specific setup of the provider.
+An additional document could be provided in the future to give basic
+guidelines for this task.
+
+A list of the default ports used in Kubernetes as well as the components accessing them can be found below:
+
+#### Control plane nodes
+
+| Ports     | Protocol | Purpose                 | Used by              | Type               |
+|-----------|----------|-------------------------|----------------------|--------------------|
+| 6443      | TCP      | API server              | All                  | External, internal |
+| 2379-2380 | TCP      | etcd server             | kube-apiserver, etcd | Internal           |
+| 10250     | TCP      | Kubelet API             | Self, Control plane  | Internal           |
+| 10255     | TCP      | Read-only Kubelet API   | (Monitoring)         | External, Internal |
+| 10257     | TCP      | kube-controller-manager | Self                 | Internal           |
+| 10259     | TCP      | kube-scheduler          | Self                 | Internal           |
+
+#### Worker nodes
+
+| Ports       | Protocol | Purpose               | Used by             | Type               |
+|-------------|----------|-----------------------|---------------------|--------------------|
+| 10250       | TCP      | Kubelet API           | Self, Control plane | Internal           |
+| 10255       | TCP      | Read-only Kubelet API | (Monitoring)        | External, internal |
+| 30000-32767 | TCP      | NodePort Services     |                     | External           |
+
+### API security, authentication and authorization
+
+In order to secure Kubernetes against bad actors, limiting and securing access to API requests
+is recommended, since requests to those are able to control the entire Kubernetes cluster.
+Access control is applied to both human users and Kubernetes service accounts, which goes through
+several stages after a request reaches the API.
+
+1. The Kubernetes API server listens on port 6443 on the first non-localhost network interface by default,
```

Comment:
Hm good point. I just took the documentation from [here](https://kubernetes.io/docs/concepts/security/controlling-access/) for granted, but the [kube-api server docs](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/) tell a different story, since `--bind-address` mentions
```
--bind-address string     Default: 0.0.0.0
	

The IP address on which to listen for the --secure-port port. The associated interface(s) must be reachable by the rest of the cluster, and by CLI/web clients. If blank or an unspecified address (0.0.0.0 or ::), all interfaces and IP address families will be used.
```
which would put the `kube-api` per default on ALL interfaces.

I guess this is somewhat contradicting and could require an upstream issue.